[FIX] website: fix rounded boxed header template megamenu issues

chgo-odoo opened the pull request:
This PR aims to solve two issues within the `rounded box` header
templates, especially in the megamenu design:

1- The megamenu is missing a border at the top. As all our header have
sharp edges, our meganumenu doesn't have a `border-top` by default to
render it correctly. This works well for all the headers except this one
as it has rounded edges.

2- The megamenu should be spaced from the header. Similar to above, our
megamenu are currently sticked to the header, for design purpose and to
ease the navigation. In the case of this specific header, it should be
spaced.

To handle these issues, we remove the border-top if we are not in the case
of the rounded box header. For the spacing issue, we rely on the navbar
padding and move the megamenu down of the same value.

task-3608124

| Master | This PR |
|--------|--------|
| ![image](https://github.com/user-attachments/assets/386bfdd6-0483-41f6-a7fc-2f27c0be3226) | ![image](https://github.com/user-attachments/assets/19a6c442-f8ad-4729-aace-c0f2c831611e) | 

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

robodoo commented:
[![Pull request status dashboard](https://mergebot.odoo.com/odoo/odoo/pull/171050.png)](https://mergebot.odoo.com/odoo/odoo/pull/171050)

chgo-odoo commented:
Hello @Brieuc-brd 👋 

Could you have a look at this small fix for the boxed header please?

Thanks 😃

Brieuc-brd commented on addons/website/static/src/scss/website.scss:
```diff
@@ -1537,6 +1537,12 @@ header {
         @include o-apply-colors('header-sales_four-custom');
         @include o-add-gradient('menu-secondary-gradient');
     }
+} @else if o-website-value('header-template') == 'boxed' {
+    // Use SCSS variable instead of utility class for this header
+    // to ease the maintainability of the code (see .o_mega_menu)
+    .navbar {
+        padding: $navbar-padding-y 0;
+    }
```
I'm not sure to understand why this is necessary 🤔 

The navbar already defines by default this variable for vertical paddings : (https://github.com/odoo/odoo/blob/master/addons/web/static/lib/bootstrap/scss/_navbar.scss#L9)

And since you handle the side paddings with an utility class, I think you can remove this part. Or did I missed something ?

chgo-odoo replied:
Ah indeed, I get your point 😅 

So currently, without any change, the padding of navbar is handled by a `py-2` utility class.

To space the mega-menu accordingly, first first that comes to my mind is to reuse the same value, but I can't define `$spacer * .5` because as soon as we'd change this `py-2` the spacing of the megamenu would be different.

----------

But you're right, I think we can just remove this part and keep the megamenu CSS only

Brieuc-brd replied:
> But you're right, I think we can just remove this part and keep the megamenu CSS only

I think you can still remove the `py-2` as you did [here](https://github.com/odoo/odoo/pull/171050/files#diff-e9f9ae4b3f4edaeb205bead270f193670b3a0b41cd97482225dd9f7137a12f9dL1460) since the default variable (`$navbar-padding-y`) has the same value 👍 
Tbh, I'm not realy sure why we forced this vertical padding with an utility classe 🤔

Brieuc-brd reviewed (changes requested):
Thanks for this fix @chgo-odoo 💪 
I've left a comment below.

Also, I saw that you first targeted `17.0` then you switched to `master`.
Can't we apply this fix to `17.0` or at least to `18.0` ?

Thanks again !

chgo-odoo commented:
@Brieuc-brd I applied the change you suggested, could you check please?

Thanks !

Brieuc-brd reviewed (commented):
LGTM, thanks ! 👍

Just a thought, since your PR fixes spacing issues for headers, could you also check  this spacing issue when the nav use the "Pills" link style in the mobile header ? Thanks in advance !
![Capture d’écran 2024-10-14 à 16 31 31](https://github.com/user-attachments/assets/7a5c3591-97cf-4ddf-8d33-ba50939b4e7d)

Also, about my last comment :

> Also, I saw that you first targeted 17.0 then you switched to master.
Can't we apply this fix to 17.0 or at least to 18.0 ?

Can we ? 🙂

chgo-odoo commented:
Skipped the last part of your message sorry ahah, we might be able to backport this yes, it'll require a lot of `!important` but we might be able to 😄 

I think we can move this task to tech review and back-port it afterward in another PR, right?

>since your PR fixes spacing issues for headers, could you also check this spacing issue when the nav use the "Pills" link style in the mobile header ?

Looks a bit out of scope as this PR only aims to fix an issue related to a specific header. But I can create a task and have a look at it, indeed 😃  wdyt?

Brieuc-brd reviewed (approved):
Ok then, you can move it to tech. review, thanks again ! 🟢

> Looks a bit out of scope as this PR only aims to fix an issue related to a specific header. But I can create a task and have a look at it, indeed 😃 wdyt?

Sure ! 👍 

> I think we can move this task to tech review and back-port it afterward in another PR, right?

I'm not sure what would be the best in this case 🤔 
I'd let @odoo/rd-website decide 🙂

chgo-odoo commented:
Hello @odoo/rd-website 👋 

I rebased this small fix as it needed to be adapted regarding https://github.com/odoo/odoo/pull/140083. 

Could you have a look at it please?
Thanks !

qsm-odoo reviewed (changes requested):
@chgo-odoo As it is 2 different fixes, could you make 2 commits? :pray: Also I think the first one (the missing padding in mobile) could be considered a bug to fix in stable, at least in 18.0 (so move that one to a PR targeting stable). Thanks :pray:

qsm-odoo commented:
@chgo-odoo You cannot change or remove things in stable XML. So this would not be possible to do:

![image](https://github.com/user-attachments/assets/0351b67c-2e2a-4775-8451-99cadcb22efe)

You can **add** the class to add paddings though.

Also from a function point of view, I don't think it is as "critical" as the missing navbar padding. People can leave with it or could even rely on the current style. Far less likely with the missing horizontal padding.

chgo-odoo commented:
@qsm-odoo I extracted the change for the padding in https://github.com/odoo/odoo/pull/186396 as discussed.

I added a screenshot for reference in both PR and updated this PR description 😃

chgo-odoo commented:
Hello @qsm-odoo, small reminder 🙏 

The change related to the navbar padding is [in this pull request](https://github.com/odoo/odoo/pull/186396).

Could you have a look at it and eventually r+ ?

Thanks !

qsm-odoo reviewed (approved):
Nice :+1: 

@robodoo r+
